### PR TITLE
Issue #13345: Enable examples tests for PackageNameCheckExamplesTest

### PR DESCRIPTION
**Pull request by Mostafahassen1**
### Issue #13345

https://checkstyle.org/checks/naming/packagename.html#Examples

@romani 

image

I have error in `azure-pipelines/ checkstyle.checkstyle`

### CODE REVIEW

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example1.java [hunk 1]
```diff
@@ -7,11 +7,14 @@
 */
 
 // xdoc section -- start
-package com; // OK
+
+
 package COM; // violation, name 'COM' must match pattern '^[a-z]+(\.[a-zA-Z_]\w*)*$'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example1.java:12]: Put here exact package name as it is folder.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example1.java [hunk 2]
```diff
@@ -7,11 +7,14 @@
...
-package com.A.checkstyle.checks; // OK
-package com.checkstyle1.checks; // OK
-package com.checkSTYLE.checks; // OK
-package com._checkstyle.checks_; // OK
+
+
+
+public class Example1 {
+
+}
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example1.java:18]: Make it `public class Example1 {}`

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>
+    </module>
+  </module>
+</module>
+*/
+
+// xdoc section -- start
+
+
+package COM; // violation, name 'COM' must match pattern '^[a-z]+(\.[a-z][a-z0-9]*)*$'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java:15]: Keep package  same as it is (like folder).
Java file should be be compiled.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="PackageName">
+      <property name="format"
+        value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java:6]: Change format to do violation on `com.puppycrawl.tools.checkstyle.checks.naming.packagename`
For example make it to demand package to start from capital letters not lowercase.

**Review by romani — CHANGES_REQUESTED** [review]
I am ok to keep single package per example.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java (hunk 4 above)
Comment (Mostafahassen1) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java:6]: do you want change this `value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>`
 to start with  capital letters  or what

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java:6]: Yes.
`^[A-Z].....`.
Just to show that user can do anything he wants

**Comment by romani** [comment]
Always single commit.
Read https://checkstyle.org/beginning_development.html#Starting_Development
Or watch videos on how to manage PR.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java (hunk 4 above)
Comment (Mostafahassen1) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java:6]: Done , but there is error again in CI/CD , I don't know what the reason 

```java

/*xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="PackageName">
      <property name="format"
        value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>
    </module>
  </module>
</module>
*/

package com.puppycrawl.tools.checkstyle.checks.naming.packagename;

// xdoc section -- start
public class Example2 {}
// xdoc section -- end
```

**Comment by romani** [comment]
`Caused by: java.nio.file.NoSuchFileException: src/xdocs-examples/resources\com\puppycrawl\tools\checkstyle\checks\naming\packagename\Example1.java`

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java (hunk 4 above)
Comment (Mostafahassen1) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java]: I made it, but it still can't pass CI/CD.

**Comment by Mostafahassen1** [comment]
Hi @romani   , i'm so sorry but i can't do understand why this happened  and how can solve it 
 `Caused by: java.nio.file.NoSuchFileException: src/xdocs-examples/resources\com\puppycrawl\tools\checkstyle\checks\naming\packagename\Example1.java`

File: src/xdocs/checks/naming/packagename.xml.template [hunk 5]
```diff
@@ -34,33 +34,35 @@
...
 
       <subsection name="Examples" id="Examples">
-        <p id="Example1-config">
+        <p id="resources\com\puppycrawl\tools\checkstyle\checks\naming\packagename\Example1-config">
           To configure the check:
         </p>
         <macro name="example">
           <param name="path"
-                 value="resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example1.txt"/>
+                 value="resources\com\puppycrawl\tools\checkstyle\checks\naming\packagename\Example1.java"/>
```
Comment (romani) [inline src/xdocs/checks/naming/packagename.xml.template:42]: Do not change `/` to `\`.
Just change extension from txt to java

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs/checks/naming/packagename.xml.template (hunk 5 above)
Comment (Mostafahassen1) [reply src/xdocs/checks/naming/packagename.xml.template]: I'm sorry, I solved this but still can't pass CI/CD.

**Comment by romani** [comment]
Please look at each failures and try to fix them  , they print what should be improved.

Squash all in one commit and pull latest sources from our repository.

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
Squash commits

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+      <property name="format"
+        value="^[A-Z]+(\.[a-z][a-z0-9]*)*$"/>
+    </module>
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.naming.packagename; // violation
+
+// xdoc section -- start
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/Example2.java:14]: Put such comments above package

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Pull request closed without merging**